streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala

```diff
@@ -34,7 +34,7 @@ object FlinkShimsProxy extends Logger {
     Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
 
   private[this] val SHIMS_PATTERN = Pattern.compile(
-    "streampark-flink-shims_flink-(1.12|1.13|1.14|1.15|1.16)_(2.11|2.12)-(.*).jar",
+    ConfigConst.STREAMPARK_FLINK_SHIMS_JAR_REGEX,
```

Comment:
I suggest add a new method, so we don’t need to change the code  when support a new flink version
```
  private[this] def getShimsLibPattern(flinkVersion: FlinkVersion): Pattern = {
    val array = flinkVersion.version.split("\\.").map(_.trim.toInt)
    val reg = array match {
      case Array(1, v, _) if v >= 12 && v <= 14 =>
        s"streampark-flink-shims_flink-1.${v}_(2.11|2.12)-(.*).jar"
      case Array(1, v, _) if v >= 15 =>
        s"streampark-flink-shims_flink-1.${v}_2.12-(.*).jar"
      case _ => throw new UnsupportedOperationException(s"Unsupported flink version: $flinkVersion")
    }
    Pattern.compile(reg, Pattern.CASE_INSENSITIVE | Pattern.DOTALL)
  }
```